Code/Tools/AssetProcessor/native/AssetManager/PathDependencyManager.cpp

```diff
@@ -342,34 +404,26 @@ namespace AssetProcessor
             }
             else
             {
-                for (const auto& productEntry : products)
-                {
-                    const AZStd::string& productName = productEntry.m_productName;
-
-                    // strip path of the leading asset platform /<platform>
-                    AZStd::string strippedPath = AssetUtilities::StripAssetPlatform(productName).toUtf8().constData();
-                    SanitizeForDatabase(strippedPath);
-
-                    if (strippedPath == matchedPath)
-                    {
-                        matchedProducts.push_back(productEntry);
-                    }
-                }
+                matchedProducts.push_back(*searchEntry->m_productEntry);
             }
 
             // Go through each dependency we're resolving and create a db entry for each product that resolved it (wildcard/source dependencies will generally create more than 1)
-            SaveResolvedDependencies(sourceEntry, exclusionMaps, sourceNameWithScanFolder, pair.second, matchedPath, isSourceDependency, matchedProducts, dependencyContainer);
+            SaveResolvedDependencies(sourceEntry, exclusionMaps, sourceNameWithScanFolder, pair.second, searchEntry->m_path, isSourceDependency, matchedProducts, dependencyContainer);
         }
 
-        // Save everything to the db
-        if (!m_stateData->UpdateProductDependencies(dependencyContainer))
+        // Copy the set back to the standard container
+        AzToolsFramework::AssetDatabase::ProductDependencyDatabaseEntryContainer dependencyVector;
+        dependencyVector.insert(dependencyVector.end(), dependencyContainer.begin(), dependencyContainer.end());
```

Comment:
You can construct a vector using two iterators.
Furthermore that those iterators can be `move_iterators`, which perform an `std::move` on each element being added to the container.
```suggestion
        AzToolsFramework::AssetDatabase::ProductDependencyDatabaseEntryContainer dependencyVector{ AZStd::make_move_iterator(dependencyContainer.begin()), AZStd::make_move_iterator(dependencyContainer.end()) };
```